components/layout/query.rs

```diff
@@ -1182,3 +1184,215 @@ fn transform_au_rectangle(
+                }
+            },
+            Position::Static => {
+                containing_block_chain.insert(target_node.opaque().0, true);
+
+                // Find the nearest ancestor that establishes a containing block for static descendants
+                for &ancestor_node in ancestor_nodes {
+                    if let Some(element) = ancestor_node.as_element() {
+                        let style_data = element.style_data();
+                        let style = style_data.styles.primary();
+                        let position = style.get_box().position;
+                        let overflow_x = style.get_box().overflow_x;
+                        let overflow_y = style.get_box().overflow_y;
+
+                        // An element establishes a containing block for static descendants if:
+                        // - It has position != static
+                        // - Or it has overflow != visible (establishes BFC)
+                        if position != Position::Static ||
+                            overflow_x != Overflow::Visible ||
+                            overflow_y != Overflow::Visible
+                        {
+                            containing_block_chain.insert(ancestor_node.opaque().0, true);
+                            break;
+                        }
+                    }
+                }
+            },
+            Position::Relative => {
+                containing_block_chain.insert(target_node.opaque().0, true);
+
+                // For relative elements, the containing block is the same as for static
+                for &ancestor_node in ancestor_nodes {
+                    if let Some(element) = ancestor_node.as_element() {
+                        let style_data = element.style_data();
+                        let style = style_data.styles.primary();
+                        let position = style.get_box().position;
+                        let overflow_x = style.get_box().overflow_x;
+                        let overflow_y = style.get_box().overflow_y;
+
+                        // An element establishes a containing block for relative descendants if:
+                        // - It has position != static
+                        // - Or it has overflow != visible (establishes BFC)
+                        if position != Position::Static ||
+                            overflow_x != Overflow::Visible ||
+                            overflow_y != Overflow::Visible
+                        {
+                            containing_block_chain.insert(ancestor_node.opaque().0, true);
+                            break;
+                        }
+                    }
+                }
+            },
+            Position::Sticky => {
+                containing_block_chain.insert(target_node.opaque().0, true);
+
+                // For sticky elements, the containing block is established by the nearest ancestor
+                // that is a scroll container or the viewport
```

Comment:
Not very familiar with sticky, but I'm quite sure this should be the same as for static and relative.